plutus-core/plutus-core/src/PlutusCore/Builtin/Meaning.hs

```diff
@@ -116,33 +128,51 @@ function and the 'TypeScheme' of the built-in function will be derived automatic
 monomorphic and simply-polymorphic cases no types need to be specified at all.
 -}
 
+type family Length xs where
+    Length '[]       = 'Z
+    Length (_ ': xs) = 'S (Length xs)
+
 type family GetArgs a :: [GHC.Type] where
     GetArgs (a -> b) = a ': GetArgs b
     GetArgs _        = '[]
 
 -- | A class that allows us to derive a monotype for a builtin.
 class KnownMonotype val args res a | args res -> a, a -> res where
     knownMonotype :: TypeScheme val args res
+    knownMonoruntime :: ReadKnownM (FoldArgsEx args, FoldArgs args res) -> BuiltinRuntime val
 
 -- | Once we've run out of term-level arguments, we return a 'TypeSchemeResult'.
 instance (res ~ res', KnownTypeAst (UniOf val) res, MakeKnown val res) =>
             KnownMonotype val '[] res res' where
     knownMonotype = TypeSchemeResult
+    knownMonoruntime getFs = BuiltinRuntimeResult $ fmap (makeKnown $ Just ()) <$> getFs
+    {-# INLINE knownMonoruntime #-}
 
 -- | Every term-level argument becomes as 'TypeSchemeArrow'.
 instance
         ( KnownTypeAst (UniOf val) arg, MakeKnown val arg, ReadKnown val arg
         , KnownMonotype val args res a
         ) => KnownMonotype val (arg ': args) res (arg -> a) where
     knownMonotype = TypeSchemeArrow knownMonotype
+    knownMonoruntime getFs =
+        BuiltinRuntimeArrow $ \mem arg ->
+            knownMonoruntime @val @args @res $ do
+                (exF, f) <- getFs
+                x <- readKnown (Just ()) arg
```

Comment:
Note to self: don't forget to drop that `Just ()` argument once unlifting is changed.